lib/foreman_hooks/util.rb

```diff
@@ -8,14 +8,26 @@ def render_hook_type
       when Host::Managed
         'host'
       else
-        self.class.name.downcase
+        self.class.name.demodulize.downcase
     end
   end
 
   def render_hook_json
     # APIv2 has some pretty good templates.  We could extend them later in special cases.
     # Wrap them in a root node for pre-1.4 compatibility
-    json = Rabl.render(self, "api/v2/#{render_hook_type.tableize}/show", :view_path => 'app/views', :format => :json)
+    paths = [File.join("app","views")]
+
+    # Foreman 1.7 added support for paths for plugin metadata.
+    # If that method exists, then use the path to expose the views
+    # for the plugins.
+    if Foreman::Plugin.method_defined? :path
+      Foreman::Plugin.registered_plugins.each do |key, plugin|
+        paths << File.join(plugin.path, "app", "views")
+        paths << File.join(plugin.path, "app", "views", key.to_s)
```

Comment:
Using the 'key' (plugin name) isn't always correct as the class namespace may be different - foreman-tasks is officially called "foreman-tasks" but uses ForemanTasks as the namespace, so app/views/foreman_tasks exists.

I'd suggest not adding the plugin name here but pulling the module name from the hook class and using that.